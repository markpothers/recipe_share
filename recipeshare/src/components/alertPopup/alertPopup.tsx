import { Modal, Text, TouchableOpacity, View } from "react-native";
import { responsiveFontSize, responsiveHeight, responsiveWidth } from "react-native-responsive-dimensions";

import Icon from "react-native-vector-icons/MaterialCommunityIcons";
import React from "react";
import { centralStyles } from "../../centralStyleSheet"; //eslint-disable-line no-unused-vars
import { styles } from "./alertPopupStyleSheet";

type Props = {
	title: string;
	close?: () => void;
	closeText?: string;
	onYes: () => void;
	yesText?: string;
};

export function AlertPopup(props: Props) {
	return (
		<Modal animationType="fade" transparent={true} visible={true}>
			<View
				style={[
					styles.modalFullScreenContainer,
					{
						height: responsiveHeight(100),
						width: responsiveWidth(100),
					},
				]}
				accessibilityRole="alert"
			>
				<View style={styles.contentsContainer}>
					<View style={styles.titleContainer}>
						<Text maxFontSizeMultiplier={1.5} style={styles.title}>
							{props.title}
						</Text>
					</View>
					<View
						style={[
							centralStyles.formSection,
							{
								width: responsiveWidth(80),
								marginTop: responsiveHeight(2),
								marginBottom: responsiveHeight(2),
							},
						]}
					>
						<View style={[centralStyles.formInputContainer, { justifyContent: "space-around" }]}>
							{props.close && (
								<TouchableOpacity
									style={centralStyles.greenRectangleButton}
									activeOpacity={0.7}
									onPress={props.close}
									testID={"closeButton"}
									accessibilityRole="button"
								>
									<Icon
										style={centralStyles.yellowButtonIcon}
										size={responsiveHeight(4)}
										name="cancel"
									></Icon>
									<Text maxFontSizeMultiplier={2} style={centralStyles.yellowButtonText}>
										{props.closeText ? props.closeText : "Cancel"}
									</Text>
								</TouchableOpacity>
							)}
							{!props.close && <View style={[centralStyles.greenRectangleButton, { opacity: 0 }]}></View>}
							<TouchableOpacity
								style={centralStyles.greenRectangleButton}
								activeOpacity={0.7}
								onPress={props.onYes}
								testID={"yesButton"}
								accessibilityRole="button"
							>
								<Icon
									style={centralStyles.yellowButtonIcon}
									size={responsiveHeight(4)}
									name="check"
								></Icon>
								<Text maxFontSizeMultiplier={2} style={centralStyles.yellowButtonText}>
									{props.yesText ? props.yesText : "Yes"}
								</Text>
							</TouchableOpacity>
						</View>
					</View>
				</View>
			</View>
		</Modal>
	);
}
