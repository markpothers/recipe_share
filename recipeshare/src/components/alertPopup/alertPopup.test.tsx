import { fireEvent, render } from "@testing-library/react-native";

import { AlertPopup } from "./alertPopup";
import React from "react";

describe("AlertPopup", () => {
	let mockClose;
	let mockYes;

	beforeEach(() => {
		mockClose = jest.fn();
		mockYes = jest.fn();
	});

	test("renders with 2 buttons and customizable text", () => {
		const { toJSON, getAllByRole } = render(
			<AlertPopup
				close={mockClose}
				closeText={"mock close text"}
				title={"Here is a mock alert question that's being asked"}
				onYes={mockYes}
				yesText={"mock yes text"}
			/>
		);
		expect(toJSON()).toMatchSnapshot();
		const buttons = getAllByRole("button");
		expect(buttons.length).toEqual(2);
	});

	test("renders with 2 buttons and default text", () => {
		const { toJSON, getAllByRole } = render(
			<AlertPopup
				close={mockClose}
				// closeText={"mock close text"}
				title={"Here is a mock alert question that's being asked"}
				onYes={mockYes}
				// yesText={"Yes button text"}
			/>
		);
		expect(toJSON()).toMatchSnapshot();
		const buttons = getAllByRole("button");
		expect(buttons.length).toEqual(2);
	});

	test("renders with 1 button when no close button and customizable text", () => {
		const { toJSON, getAllByRole } = render(
			<AlertPopup
				// close={mockClose}
				// closeText={"mock close text"}
				title={"Here is a mock alert question that's being asked"}
				onYes={mockYes}
				yesText={"Yes button text"}
			/>
		);
		expect(toJSON()).toMatchSnapshot();
		const buttons = getAllByRole("button");
		expect(buttons.length).toEqual(1);
	});

	test("calls the yes function when button pressed", () => {
		const { toJSON, getByText } = render(
			<AlertPopup
				close={mockClose}
				closeText={"mock close text"}
				title={"Here is a mock alert question that's being asked"}
				onYes={mockYes}
				yesText={"mock yes text"}
			/>
		);
		expect(toJSON()).toMatchSnapshot();
		fireEvent.press(getByText("mock yes text"));
		expect(mockYes).toHaveBeenCalled();
		expect(mockYes).toHaveBeenCalledTimes(1);
	});

	test("calls the close function when button pressed", () => {
		const { toJSON, getByText } = render(
			<AlertPopup
				close={mockClose}
				closeText={"mock close text"}
				title={"Here is a mock alert question that's being asked"}
				onYes={mockYes}
				yesText={"mock yes text"}
			/>
		);
		expect(toJSON()).toMatchSnapshot();
		fireEvent.press(getByText("mock close text"));
		expect(mockClose).toHaveBeenCalled();
		expect(mockClose).toHaveBeenCalledTimes(1);
	});
});
