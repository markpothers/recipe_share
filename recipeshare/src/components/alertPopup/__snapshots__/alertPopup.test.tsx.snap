// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlertPopup calls the close function when button pressed 1`] = `
<Modal
  animationType="fade"
  hardwareAccelerated={false}
  transparent={true}
  visible={true}
>
  <View
    accessibilityRole="alert"
    style={
      [
        {
          "alignItems": "center",
          "justifyContent": "center",
        },
        {
          "height": 1334,
          "width": 750,
        },
      ]
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#fff59b",
          "borderColor": "#104e01",
          "borderRadius": 11.25,
          "borderStyle": "solid",
          "borderWidth": 1,
          "marginLeft": "7.5%",
          "marginRight": "7.5%",
          "width": 637.5,
        }
      }
    >
      <View
        style={
          {
            "justifyContent": "center",
            "paddingTop": 13.34,
            "width": "90%",
          }
        }
      >
        <Text
          maxFontSizeMultiplier={1.5}
          style={
            {
              "color": "#104e01",
              "fontSize": 33.65552620959067,
              "fontWeight": "bold",
            }
          }
        >
          Here is a mock alert question that's being asked
        </Text>
      </View>
      <View
        style={
          [
            {
              "borderColor": "red",
              "borderWidth": 0,
              "overflow": "hidden",
              "width": "100%",
            },
            {
              "marginBottom": 26.68,
              "marginTop": 26.68,
              "width": 600,
            },
          ]
        }
      >
        <View
          style={
            [
              {
                "alignItems": "center",
                "borderColor": "purple",
                "borderRadius": 11.25,
                "borderWidth": 0,
                "flexDirection": "row",
                "flexWrap": "wrap",
                "justifyContent": "space-between",
                "marginTop": 6.67,
                "minHeight": 80.03999999999999,
                "overflow": "hidden",
                "width": "100%",
              },
              {
                "justifyContent": "space-around",
              },
            ]
          }
        >
          <View
            accessibilityRole="button"
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#104e01",
                "borderColor": "#104e01",
                "borderRadius": 11.25,
                "borderWidth": 1,
                "flexDirection": "row",
                "height": "100%",
                "justifyContent": "space-evenly",
                "maxWidth": "50%",
                "minHeight": 80.03999999999999,
                "minWidth": 285,
                "opacity": 1,
              }
            }
            testID="closeButton"
          >
            <Icon
              name="cancel"
              size={53.36}
              style={
                {
                  "color": "#fff59b",
                }
              }
            />
            <Text
              maxFontSizeMultiplier={2}
              style={
                {
                  "color": "#fff59b",
                  "fontSize": 27.53633962602873,
                  "fontWeight": "bold",
                  "paddingHorizontal": 7.5,
                  "textAlign": "center",
                }
              }
            >
              mock close text
            </Text>
          </View>
          <View
            accessibilityRole="button"
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#104e01",
                "borderColor": "#104e01",
                "borderRadius": 11.25,
                "borderWidth": 1,
                "flexDirection": "row",
                "height": "100%",
                "justifyContent": "space-evenly",
                "maxWidth": "50%",
                "minHeight": 80.03999999999999,
                "minWidth": 285,
                "opacity": 1,
              }
            }
            testID="yesButton"
          >
            <Icon
              name="check"
              size={53.36}
              style={
                {
                  "color": "#fff59b",
                }
              }
            />
            <Text
              maxFontSizeMultiplier={2}
              style={
                {
                  "color": "#fff59b",
                  "fontSize": 27.53633962602873,
                  "fontWeight": "bold",
                  "paddingHorizontal": 7.5,
                  "textAlign": "center",
                }
              }
            >
              mock yes text
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</Modal>
`;

exports[`AlertPopup calls the yes function when button pressed 1`] = `
<Modal
  animationType="fade"
  hardwareAccelerated={false}
  transparent={true}
  visible={true}
>
  <View
    accessibilityRole="alert"
    style={
      [
        {
          "alignItems": "center",
          "justifyContent": "center",
        },
        {
          "height": 1334,
          "width": 750,
        },
      ]
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#fff59b",
          "borderColor": "#104e01",
          "borderRadius": 11.25,
          "borderStyle": "solid",
          "borderWidth": 1,
          "marginLeft": "7.5%",
          "marginRight": "7.5%",
          "width": 637.5,
        }
      }
    >
      <View
        style={
          {
            "justifyContent": "center",
            "paddingTop": 13.34,
            "width": "90%",
          }
        }
      >
        <Text
          maxFontSizeMultiplier={1.5}
          style={
            {
              "color": "#104e01",
              "fontSize": 33.65552620959067,
              "fontWeight": "bold",
            }
          }
        >
          Here is a mock alert question that's being asked
        </Text>
      </View>
      <View
        style={
          [
            {
              "borderColor": "red",
              "borderWidth": 0,
              "overflow": "hidden",
              "width": "100%",
            },
            {
              "marginBottom": 26.68,
              "marginTop": 26.68,
              "width": 600,
            },
          ]
        }
      >
        <View
          style={
            [
              {
                "alignItems": "center",
                "borderColor": "purple",
                "borderRadius": 11.25,
                "borderWidth": 0,
                "flexDirection": "row",
                "flexWrap": "wrap",
                "justifyContent": "space-between",
                "marginTop": 6.67,
                "minHeight": 80.03999999999999,
                "overflow": "hidden",
                "width": "100%",
              },
              {
                "justifyContent": "space-around",
              },
            ]
          }
        >
          <View
            accessibilityRole="button"
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#104e01",
                "borderColor": "#104e01",
                "borderRadius": 11.25,
                "borderWidth": 1,
                "flexDirection": "row",
                "height": "100%",
                "justifyContent": "space-evenly",
                "maxWidth": "50%",
                "minHeight": 80.03999999999999,
                "minWidth": 285,
                "opacity": 1,
              }
            }
            testID="closeButton"
          >
            <Icon
              name="cancel"
              size={53.36}
              style={
                {
                  "color": "#fff59b",
                }
              }
            />
            <Text
              maxFontSizeMultiplier={2}
              style={
                {
                  "color": "#fff59b",
                  "fontSize": 27.53633962602873,
                  "fontWeight": "bold",
                  "paddingHorizontal": 7.5,
                  "textAlign": "center",
                }
              }
            >
              mock close text
            </Text>
          </View>
          <View
            accessibilityRole="button"
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#104e01",
                "borderColor": "#104e01",
                "borderRadius": 11.25,
                "borderWidth": 1,
                "flexDirection": "row",
                "height": "100%",
                "justifyContent": "space-evenly",
                "maxWidth": "50%",
                "minHeight": 80.03999999999999,
                "minWidth": 285,
                "opacity": 1,
              }
            }
            testID="yesButton"
          >
            <Icon
              name="check"
              size={53.36}
              style={
                {
                  "color": "#fff59b",
                }
              }
            />
            <Text
              maxFontSizeMultiplier={2}
              style={
                {
                  "color": "#fff59b",
                  "fontSize": 27.53633962602873,
                  "fontWeight": "bold",
                  "paddingHorizontal": 7.5,
                  "textAlign": "center",
                }
              }
            >
              mock yes text
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</Modal>
`;

exports[`AlertPopup renders with 1 button when no close button and customizable text 1`] = `
<Modal
  animationType="fade"
  hardwareAccelerated={false}
  transparent={true}
  visible={true}
>
  <View
    accessibilityRole="alert"
    style={
      [
        {
          "alignItems": "center",
          "justifyContent": "center",
        },
        {
          "height": 1334,
          "width": 750,
        },
      ]
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#fff59b",
          "borderColor": "#104e01",
          "borderRadius": 11.25,
          "borderStyle": "solid",
          "borderWidth": 1,
          "marginLeft": "7.5%",
          "marginRight": "7.5%",
          "width": 637.5,
        }
      }
    >
      <View
        style={
          {
            "justifyContent": "center",
            "paddingTop": 13.34,
            "width": "90%",
          }
        }
      >
        <Text
          maxFontSizeMultiplier={1.5}
          style={
            {
              "color": "#104e01",
              "fontSize": 33.65552620959067,
              "fontWeight": "bold",
            }
          }
        >
          Here is a mock alert question that's being asked
        </Text>
      </View>
      <View
        style={
          [
            {
              "borderColor": "red",
              "borderWidth": 0,
              "overflow": "hidden",
              "width": "100%",
            },
            {
              "marginBottom": 26.68,
              "marginTop": 26.68,
              "width": 600,
            },
          ]
        }
      >
        <View
          style={
            [
              {
                "alignItems": "center",
                "borderColor": "purple",
                "borderRadius": 11.25,
                "borderWidth": 0,
                "flexDirection": "row",
                "flexWrap": "wrap",
                "justifyContent": "space-between",
                "marginTop": 6.67,
                "minHeight": 80.03999999999999,
                "overflow": "hidden",
                "width": "100%",
              },
              {
                "justifyContent": "space-around",
              },
            ]
          }
        >
          <View
            style={
              [
                {
                  "alignItems": "center",
                  "backgroundColor": "#104e01",
                  "borderColor": "#104e01",
                  "borderRadius": 11.25,
                  "borderWidth": 1,
                  "flexDirection": "row",
                  "height": "100%",
                  "justifyContent": "space-evenly",
                  "maxWidth": "50%",
                  "minHeight": 80.03999999999999,
                  "minWidth": 285,
                },
                {
                  "opacity": 0,
                },
              ]
            }
          />
          <View
            accessibilityRole="button"
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#104e01",
                "borderColor": "#104e01",
                "borderRadius": 11.25,
                "borderWidth": 1,
                "flexDirection": "row",
                "height": "100%",
                "justifyContent": "space-evenly",
                "maxWidth": "50%",
                "minHeight": 80.03999999999999,
                "minWidth": 285,
                "opacity": 1,
              }
            }
            testID="yesButton"
          >
            <Icon
              name="check"
              size={53.36}
              style={
                {
                  "color": "#fff59b",
                }
              }
            />
            <Text
              maxFontSizeMultiplier={2}
              style={
                {
                  "color": "#fff59b",
                  "fontSize": 27.53633962602873,
                  "fontWeight": "bold",
                  "paddingHorizontal": 7.5,
                  "textAlign": "center",
                }
              }
            >
              Yes button text
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</Modal>
`;

exports[`AlertPopup renders with 2 buttons and customizable text 1`] = `
<Modal
  animationType="fade"
  hardwareAccelerated={false}
  transparent={true}
  visible={true}
>
  <View
    accessibilityRole="alert"
    style={
      [
        {
          "alignItems": "center",
          "justifyContent": "center",
        },
        {
          "height": 1334,
          "width": 750,
        },
      ]
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#fff59b",
          "borderColor": "#104e01",
          "borderRadius": 11.25,
          "borderStyle": "solid",
          "borderWidth": 1,
          "marginLeft": "7.5%",
          "marginRight": "7.5%",
          "width": 637.5,
        }
      }
    >
      <View
        style={
          {
            "justifyContent": "center",
            "paddingTop": 13.34,
            "width": "90%",
          }
        }
      >
        <Text
          maxFontSizeMultiplier={1.5}
          style={
            {
              "color": "#104e01",
              "fontSize": 33.65552620959067,
              "fontWeight": "bold",
            }
          }
        >
          Here is a mock alert question that's being asked
        </Text>
      </View>
      <View
        style={
          [
            {
              "borderColor": "red",
              "borderWidth": 0,
              "overflow": "hidden",
              "width": "100%",
            },
            {
              "marginBottom": 26.68,
              "marginTop": 26.68,
              "width": 600,
            },
          ]
        }
      >
        <View
          style={
            [
              {
                "alignItems": "center",
                "borderColor": "purple",
                "borderRadius": 11.25,
                "borderWidth": 0,
                "flexDirection": "row",
                "flexWrap": "wrap",
                "justifyContent": "space-between",
                "marginTop": 6.67,
                "minHeight": 80.03999999999999,
                "overflow": "hidden",
                "width": "100%",
              },
              {
                "justifyContent": "space-around",
              },
            ]
          }
        >
          <View
            accessibilityRole="button"
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#104e01",
                "borderColor": "#104e01",
                "borderRadius": 11.25,
                "borderWidth": 1,
                "flexDirection": "row",
                "height": "100%",
                "justifyContent": "space-evenly",
                "maxWidth": "50%",
                "minHeight": 80.03999999999999,
                "minWidth": 285,
                "opacity": 1,
              }
            }
            testID="closeButton"
          >
            <Icon
              name="cancel"
              size={53.36}
              style={
                {
                  "color": "#fff59b",
                }
              }
            />
            <Text
              maxFontSizeMultiplier={2}
              style={
                {
                  "color": "#fff59b",
                  "fontSize": 27.53633962602873,
                  "fontWeight": "bold",
                  "paddingHorizontal": 7.5,
                  "textAlign": "center",
                }
              }
            >
              mock close text
            </Text>
          </View>
          <View
            accessibilityRole="button"
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#104e01",
                "borderColor": "#104e01",
                "borderRadius": 11.25,
                "borderWidth": 1,
                "flexDirection": "row",
                "height": "100%",
                "justifyContent": "space-evenly",
                "maxWidth": "50%",
                "minHeight": 80.03999999999999,
                "minWidth": 285,
                "opacity": 1,
              }
            }
            testID="yesButton"
          >
            <Icon
              name="check"
              size={53.36}
              style={
                {
                  "color": "#fff59b",
                }
              }
            />
            <Text
              maxFontSizeMultiplier={2}
              style={
                {
                  "color": "#fff59b",
                  "fontSize": 27.53633962602873,
                  "fontWeight": "bold",
                  "paddingHorizontal": 7.5,
                  "textAlign": "center",
                }
              }
            >
              mock yes text
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</Modal>
`;

exports[`AlertPopup renders with 2 buttons and default text 1`] = `
<Modal
  animationType="fade"
  hardwareAccelerated={false}
  transparent={true}
  visible={true}
>
  <View
    accessibilityRole="alert"
    style={
      [
        {
          "alignItems": "center",
          "justifyContent": "center",
        },
        {
          "height": 1334,
          "width": 750,
        },
      ]
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#fff59b",
          "borderColor": "#104e01",
          "borderRadius": 11.25,
          "borderStyle": "solid",
          "borderWidth": 1,
          "marginLeft": "7.5%",
          "marginRight": "7.5%",
          "width": 637.5,
        }
      }
    >
      <View
        style={
          {
            "justifyContent": "center",
            "paddingTop": 13.34,
            "width": "90%",
          }
        }
      >
        <Text
          maxFontSizeMultiplier={1.5}
          style={
            {
              "color": "#104e01",
              "fontSize": 33.65552620959067,
              "fontWeight": "bold",
            }
          }
        >
          Here is a mock alert question that's being asked
        </Text>
      </View>
      <View
        style={
          [
            {
              "borderColor": "red",
              "borderWidth": 0,
              "overflow": "hidden",
              "width": "100%",
            },
            {
              "marginBottom": 26.68,
              "marginTop": 26.68,
              "width": 600,
            },
          ]
        }
      >
        <View
          style={
            [
              {
                "alignItems": "center",
                "borderColor": "purple",
                "borderRadius": 11.25,
                "borderWidth": 0,
                "flexDirection": "row",
                "flexWrap": "wrap",
                "justifyContent": "space-between",
                "marginTop": 6.67,
                "minHeight": 80.03999999999999,
                "overflow": "hidden",
                "width": "100%",
              },
              {
                "justifyContent": "space-around",
              },
            ]
          }
        >
          <View
            accessibilityRole="button"
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#104e01",
                "borderColor": "#104e01",
                "borderRadius": 11.25,
                "borderWidth": 1,
                "flexDirection": "row",
                "height": "100%",
                "justifyContent": "space-evenly",
                "maxWidth": "50%",
                "minHeight": 80.03999999999999,
                "minWidth": 285,
                "opacity": 1,
              }
            }
            testID="closeButton"
          >
            <Icon
              name="cancel"
              size={53.36}
              style={
                {
                  "color": "#fff59b",
                }
              }
            />
            <Text
              maxFontSizeMultiplier={2}
              style={
                {
                  "color": "#fff59b",
                  "fontSize": 27.53633962602873,
                  "fontWeight": "bold",
                  "paddingHorizontal": 7.5,
                  "textAlign": "center",
                }
              }
            >
              Cancel
            </Text>
          </View>
          <View
            accessibilityRole="button"
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#104e01",
                "borderColor": "#104e01",
                "borderRadius": 11.25,
                "borderWidth": 1,
                "flexDirection": "row",
                "height": "100%",
                "justifyContent": "space-evenly",
                "maxWidth": "50%",
                "minHeight": 80.03999999999999,
                "minWidth": 285,
                "opacity": 1,
              }
            }
            testID="yesButton"
          >
            <Icon
              name="check"
              size={53.36}
              style={
                {
                  "color": "#fff59b",
                }
              }
            />
            <Text
              maxFontSizeMultiplier={2}
              style={
                {
                  "color": "#fff59b",
                  "fontSize": 27.53633962602873,
                  "fontWeight": "bold",
                  "paddingHorizontal": 7.5,
                  "textAlign": "center",
                }
              }
            >
              Yes
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</Modal>
`;
